feat: New meeting type carousel

igorlesnenko opened the pull request:
Fixes #5605

## Demo

https://user-images.githubusercontent.com/466991/181078228-9d0797a7-133e-447e-b963-6c7e8c569dfa.mov


https://user-images.githubusercontent.com/466991/181078239-9c0d0b28-cde5-4104-ba71-8211b1089a0d.mov

## Testing scenarios

- Open create meeting dialog and try to create various type of metings
- Do the same on mobile

BartoszJarocki commented on packages/client/components/NewMeetingCarousel.tsx:
```diff
@@ -0,0 +1,148 @@
+    alignItems: 'center',
+    padding: '20px 0'
+  },
+  marginTop: -12
```
+1 imo it looks better without that, especially on mobile

BartoszJarocki commented on packages/client/components/NewMeetingCarousel.tsx:
```diff
@@ -0,0 +1,148 @@
+  retrospective: 'Retrospective',
+  action: 'Team Check-in',
+  poker: 'Sprint Poker',
+  teamPrompt: 'Stand-Up'
```
-1 the title is `Async Standup`

BartoszJarocki commented on packages/client/components/NewMeetingCarousel.tsx:
```diff
@@ -0,0 +1,148 @@
+  retrospective: 'Improve how you work',
+  action: 'Sync up, unblock each other',
+  poker: 'Estimate effort, de-risk execution',
+  teamPrompt: 'Stand-Up'
```
-1 i guess we need a copy for that

igorlesnenko replied:
Yeah, I temporary put this here. What would be correct description? cc @enriquesanchez

BartoszJarocki commented on packages/client/components/NewMeetingCarousel.tsx:
```diff
@@ -0,0 +1,148 @@
+  color: PALETTE.WHITE,
+  fontSize: 12,
+  [MEDIA_QUERY_FUZZY_TABLET]: {
+    fontSize: 11
```
+1 this seems to be pretty small from accessibility pov, can we stick to at least 12?

igorlesnenko replied:
Made 12px

BartoszJarocki reviewed (changes requested):
looks really nice! there's only one thing that's blocking IMO, standup related titles and description

enriquesanchez reviewed (changes requested):
@igorlesnenko This is coming along very nice! 🙌 

One change I'd like to see: if I select the last meeting type (in this case Standups) the carousel should automatically scroll so the whole square is in view.

You can see what I mean around 5 seconds on the video you shared:

<img width="780" alt="Captura de Pantalla 2022-07-27 a la(s) 10 50 25 a m" src="https://user-images.githubusercontent.com/3276087/181292358-23ec95c7-4467-4d57-ab93-da3455fdb971.png">

...where Standups is selected but no fully in view. Does that make sense?

As for the description line, let's go with "Share updates, on your time"

enriquesanchez commented:
@igorlesnenko I also need to look at using a different background color for the standups square. The light shade of green we currently have does not provide enough contrast and the white text is harder to read. Let me find a better color and will update here asap.

enriquesanchez commented:
Hi @igorlesnenko 👋 

Please also change the background color of the Async Standup meeting type to `#66BC8C` (Jade 400). We are also going to show the meetings in the following order: `Retro, Standup, Poker, Check-in`

[There's more context on this Slack convo here](https://parabol.slack.com/archives/C836NA350/p1658952450835159).


Do we need to handle on a separate PR the background color change for standups that also needs to be done on the `/meetings` view?

igorlesnenko commented:
@enriquesanchez 
> We are also going to show the meetings in the following order: Retro, Standup, Poker, Check-in

I changed the default order. I just want to note that I currently put the last creating meeting for the team first in the list.

Changed color for async standup everywhere in the code

BartoszJarocki reviewed (approved):
Looks good! I'd prefer to have it full screen on mobile but that could be discussed separately, great work @igorlesnenko 👌

BartoszJarocki commented:
@Dschoordsch do you mind taking a look? thanks!

Dschoordsch reviewed (changes requested):
There some smaller UI issues, like there is a height variation between different meeting types
![NewMeetingCrousselHeightVariation](https://user-images.githubusercontent.com/7331043/181499860-3a231463-9f4f-4f3b-b001-9ef5b6510802.gif)

igorlesnenko commented on packages/client/components/NewMeetingCarousel.tsx:
```diff
@@ -0,0 +1,142 @@
+        breakpoints={{
+          [Breakpoint.FUZZY_TABLET]: {
+            slidesPerView: moreThanThreeSlides ? 3.5 : 3.15,
+            slideToClickedSlide: false,
```
I felt that it would be better to only slide if card is not fully visible. But maybe fine, enabled now.

Dschoordsch replied:
Yes, that would be better, but it wouldn't slide partially visible cards into view for me before.

igorlesnenko commented:
Merging, will tweak designs in next PRs if needed
